### PR TITLE
Various modernizations in humminbird.

### PR DESCRIPTION
**Pull request by robertlipe**
- Make humminbird_icons[] a QStringList.
 - Various attempts to reduce extra allocations and hope that Qt template magic can
 - reduce some of those pointer/len buffer move when deailing with those fixed (not zero-terminated) buffers.
 - Try to more clearly mark when arrays represent fixed buffers on device files. I don't think the icon finder ever actually worked the way it was described in the comments.

### CODE REVIEW

File: humminbird.h [hunk 1]
```diff
@@ -48,7 +49,7 @@ class HumminbirdBase
 
   /* Constants */
 
-  static constexpr const char* humminbird_icons[] = {
+  QStringList humminbird_icons = {
```
Comment (tsteven4) [inline humminbird.h]: I would recommend this be a const QStringList.  That avoids any checks for detachment as well as alleviates the need to use qAsConst with this list, so you save the two uses you added in humminbird.cc as well.

File: humminbird.cc [hunk 2]
```diff
@@ -245,25 +247,22 @@ HumminbirdBase::humminbird_read_wpt(gbfile* fin)
 
   auto* wpt = new Waypoint;
 
-  // Could probably find a way to eliminate the alloc/copy.
-  char* s = xstrndup(w.name, sizeof(w.name));
-  wpt->shortname = s;
-  xfree(s);
+  QByteArray ba = QByteArray(w.name, sizeof(w.name));
+  wpt->shortname = QString(ba).remove(QChar::Null);
```
Comment (tsteven4) [inline humminbird.cc]: Can this be
auto ba = QByteArray(w.name, qstrnlen(w.name, sizeof(w.name));
?
If so there are 2 more occurrences of this pattern below.

Comment (tsteven4) [reply humminbird.cc]: or better
wpt->shortname = QByteArray(w.name, qstrnlen(w.name, sizeof(w.name));

Comment (robertlipe) [reply humminbird.cc]: Yes! That case of having NO terminator kept haunting me. I wasn't happen with what I landed on (too many type conversions, too many many iterations over the buffer) but this is better.
 
I actually looked for a qnstrlen but not a qstrnlen. I see now you used this basic approach in garmin.cc where we have that same edge between fixed length, not terminated, buffers and QStrings. Experience!

Comment (robertlipe) [reply humminbird.cc]: This was better handled in the trunk, so I've accepted the trunk's version.

This PR has had a hard life. (owing to my short attention span) I think you and I have independtly recreated the meat of this a couple of times.

**Pull request merged**